### PR TITLE
test(runtime) #4623: file permission when running the file_editor

### PR DESCRIPTION
**Pull request by xingyaoww**
…erent users

**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Add a test case for `file_editor` when running with different user inside runtime.

---
**Link of any specific issues this addresses**

Fix https://github.com/All-Hands-AI/OpenHands/issues/4623

### CODE REVIEW

File: tests/runtime/test_ipython.py [hunk 1]
```diff
@@ -232,3 +232,87 @@ def test_ipython_package_install(temp_dir, runtime_cls, run_as_openhands):
...
+    logger.info(obs, extra={'msg_type': 'OBSERVATION'})
+    assert obs.exit_code == 0
+
+    # Try to view the file as openhands user - should fail with permission denied
+    test_code = "print(file_editor(command='view', path='/root/test.txt'))"
+    action = IPythonRunCellAction(code=test_code)
+    logger.info(action, extra={'msg_type': 'ACTION'})
+    obs = runtime.run_action(action)
+    logger.info(obs, extra={'msg_type': 'OBSERVATION'})
+    assert 'Permission denied' in obs.content
```
Comment (enyst) [inline tests/runtime/test_ipython.py:259]: This is curious, it failed in CI 🤔
image

Comment (xingyaoww) [reply tests/runtime/test_ipython.py:257]: ahhh it fails when we are running the test *as openhands*

Comment (enyst) [reply tests/runtime/test_ipython.py:257]: Oh, hah! 😅

Comment (xingyaoww) [reply tests/runtime/test_ipython.py:257]: works now :)

**Review by enyst — APPROVED** [review]
🥇

**Pull request merged**